src/lua/api-gateway/tracking/factory.lua

```diff
@@ -37,9 +37,33 @@ local cjson = require "cjson"
 --   POST /tracking
 local function _API_POST_Handler()
     local trackingManager = ngx.apiGateway.tracking.manager
-    ngx.req.read_body()
     if ( ngx.req.get_method() == "POST" ) then
        local json_string = ngx.req.get_body_data()
+
+       -- Check http://openresty-reference.readthedocs.io/en/latest/Lua_Nginx_API/#ngxreqget_body_data for details
+       if (json_string == nil) then
```

Comment:
if this line is kept under get_body_data it will always be null because nobody reads the body

Reply:
I'll update the code.